healthcare/dicomweb_study_search.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START healthcare_search_studies]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	healthcare "google.golang.org/api/healthcare/v1beta1"
+)
+
+// Create a struct to append the query parameter because the library does not currently
+// natively support this.
+type queryParam struct {
+	key, value string
+}
+
+func (qp queryParam) Get() (string, string) { return qp.key, qp.value }
+
+// dicomWebSearchStudies refines a DICOMweb studies search by appending DICOM tags to the request.
+func dicomWebSearchStudies(w io.Writer, projectID, location, datasetID, dicomStoreID, dicomWebPath string) error {
+	// projectID := "my-project"
+	// location := "us-central1"
+	// datasetID := "my-dataset"
+	// dicomStoreID := "my-dicom-store"
+	// dicomWebPath := "studies"
+	ctx := context.Background()
+
+	healthcareService, err := healthcare.NewService(ctx)
+	if err != nil {
+		return fmt.Errorf("healthcare.NewService: %v", err)
+	}
+
+	storesService := healthcareService.Projects.Locations.Datasets.DicomStores
+
+	name := fmt.Sprintf("projects/%s/locations/%s/datasets/%s/dicomStores/%s", projectID, location, datasetID, dicomStoreID)
+
+	call := storesService.SearchForStudies(name, dicomWebPath)
+	// Refine your search by appending DICOM tags to the
+	// request in the form of query parameters. This sample
+	// searches for studies containing a patient's name.
+	resp, err := call.Do(queryParam{"PatientName", "Sally Zhang"})
+	if err != nil {
+		return fmt.Errorf("Get: %v", err)
+	}
+
+	fmt.Fprintf(w, "Found studies: %+v\n", resp)
```

Comment:
Should we print this more nicely since it's just an http.Response (https://godoc.org/net/http#Response)? Any expectations about the format? Check the status code?

Reply:
Done.